packages/firestore/src/remote/remote_store.ts

```diff
@@ -424,7 +424,10 @@ export class RemoteStore implements TargetMetadataProvider {
 
       // Clear the resume token for the query, since we're in a known mismatch
       // state.
-      queryData.resumeToken = emptyByteString();
+      this.listenTargets[targetId] = queryData.copy({
+        snapshotVersion: queryData.snapshotVersion,
```

Comment:
If you follow my suggestion above, this just becomes ` { resumeToken: empyByteString() }`.